### PR TITLE
✨(frontend) allow public or private room

### PR DESCRIPTION
**Pull request by PanchoutNathan**
## Purpose

Currently, when we create a room, this room is automatically private. We must be able to have the choice to make this room public or private. 

## Proposal

We add a switch to the room settings page.

### CODE REVIEW

File: src/frontend/magnify/packages/components/.storybook/preview.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import * as React from 'react';
 import {initialize, mswDecorator} from 'msw-storybook-addon';
 import {handlers} from "../src/mocks/handlers";
 import '@fontsource/roboto'; // Defaults to weight 400.
 
+
 // Initialize MSW
 initialize();
+HttpService.init(TESTING_CONF.API_URL);
+
```
Comment (jbpenrath) [inline src/frontend/magnify/packages/components/.storybook/preview.tsx:11]:
```suggestion
// Initialize MSW
initialize();
HttpService.init(TESTING_CONF.API_URL);
```

File: src/frontend/magnify/packages/components/src/components/rooms/RoomConfig/RoomConfig.tsx [hunk 2]
```diff
@@ -141,6 +160,7 @@ export const RoomConfig = ({ room }: RoomConfigProps) => {
     startAudioMuted: room?.configuration?.startAudioMuted ?? false,
     startWithVideoMuted: room?.configuration?.startWithVideoMuted ?? true,
     screenSharingEnabled: room?.configuration?.screenSharingEnabled ?? true,
+    is_public: room?.is_public ?? false,
```
Comment (jbpenrath) [inline src/frontend/magnify/packages/components/src/components/rooms/RoomConfig/RoomConfig.tsx:163]: According to the `Room` type `is_public` cannot be null.

File: src/frontend/magnify/packages/components/src/factories/rooms/index.ts [hunk 3]
```diff
@@ -1,14 +1,15 @@
 import { faker } from '@faker-js/faker';
 import { Room, RoomAccessRole, RoomUserAccesses } from '../../types';
 
-export default function createRandomRoom(isAdmin: boolean = true): Room {
+export function createRandomRoom(isAdmin: boolean = true): Room {
```
Comment (jbpenrath) [inline src/frontend/magnify/packages/components/src/factories/rooms/index.ts:4]: Could be good to move this modification and implied updates in a dedicated commit to remove some noise.

File: src/frontend/magnify/apps/magnify-site/src/views/rooms/settings/index.test.tsx [hunk 4]
```diff
@@ -26,7 +26,33 @@ describe('RoomSettingsView', () => {
...
 
-    render(<TestingContainer router={router} />);
-    await screen.findByText('Room settings');
+    await render(<TestingContainer router={router} />);
+    await screen.findByText('Settings');
+    screen.getByText('Room settings');
+    screen.getByText('Members');
+  });
+
+  it('renders privfate room', async () => {
```
Comment (jbpenrath) [inline src/frontend/magnify/apps/magnify-site/src/views/rooms/settings/index.test.tsx:35]:
```suggestion
  it('renders public room', async () => {
```

File: src/frontend/magnify/apps/magnify-site/src/views/rooms/settings/index.test.tsx [hunk 5]
```diff
@@ -26,7 +26,33 @@ describe('RoomSettingsView', () => {
...
+          element: <RoomSettingsView />,
+        },
+      ],
+      { initialEntries: ['/rooms/1234/settings'], initialIndex: 1 },
+    );
+
+    await render(<TestingContainer router={router} />);
+    await screen.findByText('Settings');
+    screen.getByText('Room settings');
+    expect(screen.queryByText('Members')).toBe(null);
```
Comment (jbpenrath) [inline src/frontend/magnify/apps/magnify-site/src/views/rooms/settings/index.test.tsx:56]: If the room is public, we should expect that RoomUsersConfig has not been rendered.

Comment (PanchoutNathan) [reply src/frontend/magnify/apps/magnify-site/src/views/rooms/settings/index.test.tsx:56]: Yes, this is the point for this test, can you explain ?

**Comment by PanchoutNathan** [comment]
> By default a room should be public
it will be done in another PR

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**